extension/src/heartbeat_agg.rs

```diff
@@ -395,6 +395,84 @@ pub fn uptime(agg: HeartbeatAgg<'static>) -> Interval {
     agg.sum_live_intervals().into()
 }
 
+#[pg_extern]
+pub fn average_downtime(agg: HeartbeatAgg<'static>) -> Option<f64> {
+    (agg.end_time - agg.start_time - agg.sum_live_intervals()).into() / agg.num_intervals
+}
+
+#[pg_extern]
+pub fn average_uptime(agg: HeartbeatAgg<'static>) -> Option<f64> {
+    (agg.sum_live_intervals()).into() / agg.num_intervals
+}
+
+#[pg_extern]
+pub fn stddev_downtime(agg: HeartbeatAgg<'static>) -> Option<f64> {
+    // Dead ranges are the opposite of the intervals stored in the aggregate
+    let mut starts = agg.interval_ends.clone().into_vec();
+    let mut ends = agg.interval_starts.clone().into_vec();
+
+    // Fix the first point depending on whether the aggregate starts in a live or dead range
+    if ends[0] == agg.start_time {
+        ends.remove(0);
+    } else {
+        starts.insert(0, agg.start_time);
+    }
+
+    // Fix the last point depending on whether the aggregate starts in a live or dead range
+    if *starts.last().unwrap() == agg.end_time {
+        starts.pop();
+    } else {
+        ends.push(agg.end_time);
+    }
+
+    let mut data = vec![];
+    for i in 0..agg.num_intervals as usize {
+        vec![].push(ends[i] - starts[i]);
```

Comment:
I think you meant this?
```suggestion
        data.push(ends[i] - starts[i]);
```